tests/RobotFramework/tests/tedge/certificate_signing_request.robot

```diff
@@ -0,0 +1,57 @@
+*** Settings ***
+Resource            ../../resources/common.resource
+Library             ThinEdgeIO
+
+Test Teardown       Get Logs
+
+Test Tags           theme:cli
+
+
+*** Test Cases ***
+Generate CSR using the device-id from an existing certificate and private key
```

Comment:
This test is already nicely complete. To make it comprehensive, you can also check that `tedge cert create-csr` doesn't alter neither the private key nor the certificate, when existing.